CONTRIBUTING.md

```diff
@@ -17,6 +17,45 @@ These are, in general, rules that you should be following while contributing to
 - Issues would be assigned on a "First Come, First Served" basis.
 - Do mention (@maintainer) the project maintainer if your PR isn't reviewed within a few days.
 
+## Build Instructions
+
+- Fork & clone the project (please see below for instructions on how to do this).
+- Setup a virtual environment:
+```
+python3 -m venv .venv
+source .venv/bin/activate
+```
+- Install required packages
+
+```
+pip install pre-commit poetry maturin patchelf
```

Comment:
Hey @VishnuSanal , why do we need`patchelf` ?

Reply:
@VishnuSanal , I did some research. Looks like this a Linux specific dependency issue. More of arch and related distros.

Could you mention something like - “If you face a patchelf related issue”, install this? Or something similar on these grounds?

Reply:
@VishnuSanal , no no. We should definitely keep it. Just specify as a fyi somewhere